bundles/CoreBundle/Resources/migrations/Version20181115114003.php

```diff
@@ -0,0 +1,143 @@
+                unset($customLayout->fieldDefinitions);
+                $customLayout->setLayoutDefinitions($layoutDefinition);
+
+                $phpFile = $this->updateCustomLayoutDefinitionToPhp($customLayout);
+
+                if (is_file($phpFile)) {
+                    @unlink($psfFil);
+                }
+            }
+        }
+    }
+
+    public function updateCustomLayoutDefinitionToPhp($customLayout)
+    {
+
+        // save definition as a php file
+        $definitionFile = $this->getDefinitionFile($customLayout->getId());
+        if (!is_writable(dirname($definitionFile)) || (is_file($definitionFile) && !is_writable($definitionFile))) {
+            throw new \Exception(
+                'Cannot write definition file in: '.$definitionFile.' please check write permission on this directory.'
+            );
+        }
+
+        try {
+            $infoDocBlock = $this->getInfoDocBlock();
+            $clone = clone $customLayout;
+            $clone->setDao(null);
+            unset($clone->fieldDefinitions);
+
+            self::cleanupForExport($clone->layoutDefinitions);
+
+            $exportedCustomLayout = var_export($clone, true);
+
+            $data = '<?php ';
+            $data .= "\n\n";
+            $data .= $infoDocBlock;
+            $data .= "\n\n";
+
+            $data .= "\nreturn ".$exportedCustomLayout.";\n";
+
+            \Pimcore\File::putPhpFile($definitionFile, $data);
+
+            return $definitionFile;
+
+        } catch (\Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @param null $name
+     *
+     * @return string
+     */
+    public function getDefinitionFile($layoutId)
+    {
+        $file = PIMCORE_CUSTOMLAYOUT_DIRECTORY.'/custom_definition_'. $layoutId .'.php';
+
+        return $file;
+    }
+
+    /**
+     * @return string
+     */
+    public function getInfoDocBlock()
+    {
+        $cd = '';
+
+        $cd .= '/** ';
+        $cd .= "\n";
+        $cd .= '* Generated at: '.date('c')."\n";
+
+        $cd .= '* Changed by: system (0)'."\n";
+
+        if (isset($_SERVER['REMOTE_ADDR'])) {
+            $cd .= '* IP: '.$_SERVER['REMOTE_ADDR']."\n";
+        }
+
+        $cd .= '*/ ';
+
+        return $cd;
+    }
+
+    /**
+     * @param $data
+     */
+    public static function cleanupForExport(&$data)
+    {
+        if (isset($data->fieldDefinitionsCache)) {
+            unset($data->fieldDefinitionsCache);
+        }
+
+        if (method_exists($data, 'getChilds')) {
+            $children = $data->getChilds();
+            if (is_array($children)) {
+                foreach ($children as $child) {
+                    self::cleanupForExport($child);
+                }
+            }
+        }
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+        // this down() migration is auto-generated, please modify it to your needs
```

Comment:
what if someone creates alphanumeric ids(definable identifier) and then reverting id column to integer is not possible with existing data?

Reply:
@brusch It is migration script problem as it will throw error for incorrect integer values when you try to revert column 'id' from varchar to int (if it has alphanumeric ids).

Reply:
@dvesh3 i got the point! But we could revert all the numeric ones, right? 
For a typical downgrade case (right after the upgrade because something is not behaving correctly), that would be ok, or what do you think?